lib/dpl/provider/pages.rb

```diff
@@ -60,17 +70,32 @@ def github_deploy
         context.shell "echo '#{@gh_fqdn}' > CNAME" if @gh_fqdn
         context.shell 'git add .'
         context.shell "FILES=\"`git commit -m 'Deploy #{@project_name} to #{@gh_ref}:#{@target_branch}' | tail`\"; echo \"$FILES\"; echo \"$FILES\" | [ `wc -l` -lt 10 ] || echo '...'"
-        context.shell "git push --force --quiet 'https://#{@gh_token}@#{@gh_ref}' master:#{@target_branch} > /dev/null 2>&1"
+        if @gh_token
+          target_repo = "https://#{@gh_token}@#{@gh_ref}"
+        else
+          # Depend on the key set up in push_app.
+          target_repo = "git@github.com:#{slug}.git"
```

Comment:
It's not always `github.com` and `@gh_ref = "#{@gh_url}/#{slug}.git"` (see above). So you could probably do smth like:

```ruby
url_start = 'git' if ssh_key is given else "https://#{@gh_token}"
url_delimiter = ':' if ssh_key is given else '/'
target_repo = "#{url_start}@#{@gh_url}#{url_delimiter}#{slug}.git"
```

But I'd store `@gh_ref` in init.